primefaces/src/main/java/org/primefaces/behavior/confirm/ConfirmBehavior.java

```diff
@@ -78,7 +81,26 @@ public String getScript(ClientBehaviorContext behaviorContext) {
         String source = getSource();
         String type = JSONObject.quote(getType());
         String headerText = JSONObject.quote(getHeader());
-        String messageText = JSONObject.quote(getMessage());
+
+        String messageText;
+        UIComponent messageFacetComponent = component.getFacet("confirmMessage");
+        if (null != messageFacetComponent) {
+            ResponseWriter originalWriter = context.getResponseWriter();
+            FastStringWriter fsw = new FastStringWriter();
+            ResponseWriter clonedWriter = originalWriter.cloneWithWriter(fsw);
+            context.setResponseWriter(clonedWriter);
+            try {
+                messageFacetComponent.encodeAll(context);
+            }
+            catch (IOException ex) {
+                fsw.write("error encoding confirm message facet");
```

Comment:
I think we typically just let these kinds of errors just bubble up?

Reply:
Actually use `ComponentUtils.encodeComponent` which wraps a lot of this logic up nicely!

Reply:
Thanks for the review. Indeed nice utility functions. Will use these.